scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -1015,9 +1015,10 @@ def _create_build_dir_link(self, links_dir_path, instance):
         self.link_dir_counter += 1
 
 
-def change_skip_to_error_if_integration(options, instance):
-    ''' If integration mode is on all skips on integration_platforms are treated as errors.'''
-    if options.integration and instance.platform.name in instance.testsuite.integration_platforms \
+def change_skip_to_error_on_integration_and_allow_platforms(options, instance):
```

Comment:
need a shorter function name, what if we wanted to add arch_allow to the check? :)